gnomad/utils/constraint.py

```diff
@@ -125,3 +127,274 @@ def collapse_strand(
+    :param possible_variants: The annotation name of the possible variant counts for each combination of keys in `ht`.
+    :param weighted: Whether to generalize the model to weighted least squares using 'possible_variants', defaults to False.
+    :return: A Dictionary of intercepts and slopes for a plateau model.
+    """
+    agg_expr = {
+        "total": hl.agg.group_by(
+            ht[cpg],
+            hl.agg.linreg(
+                ht[observed_variants] / ht[possible_variants],
+                [1, ht[mu_snp]],
+                weight=ht[possible_variants] if weighted else None,
+            ).beta,
+        )
+    }
+    return ht.aggregate(hl.struct(**agg_expr))
+
+
+def build_coverage_model(
+    ht: hl.Table,
+    low_coverage_obs_exp: str,
+    log_coverage: str,
+) -> Tuple[float, float]:
+    """
+    Build coverage model.
+
+    This function uses linear regression to build a model of log10(coverage) to this scaled ratio as a correction
+    factor for predicted proportion observed/expected variant counts at low coverage sites.
+
+    The x and y of the coverage model:
+        - x: log10('exome_coverage') at low coverage site
+        - y: sum('observed_variants')/ (`high_coverage_scale_factor` * sum('possible_variants' * 'mu_snp') at low coverage site
+            where `high_coverage_scale_factor` = sum('observed_variants') / sum('possible_variants' * 'mu_snp') at high coverage site
+
+    :param ht: Low coverage Table.
+    :param low_cov_oe_expr: The annotation name of observed:expected ratio for a given coverage level.
+    :param log_cov_expr: The annotation name of log10 coverage.
+    :return: Tuple with intercept and slope of the model.
+    """
+    return tuple(
+        ht.aggregate(
+            hl.agg.linreg(ht[low_coverage_obs_exp], [1, ht[log_coverage]])
+        ).beta
+    )
+
+
+def get_all_pop_lengths(
+    ht, pops: Tuple[str], prefix: str = "observed_"
+) -> List[Tuple[str, str]]:
+    """
+    Get the minimum array length of variant counts array in downsampings for specific per population annotations in `ht`.
+
+    The minimum array length will be used to determine the number of plateau models built for each population.
+
+    The annotations are specified by the combination of `prefix` and each population in `pops`.
+
+    :param ht: Input Table used to build population plateau models.
+    :param pops: List of populations.
+    :param prefix: Prefix of population variant count. Defaults to 'observed_'.
+    :return: A Dictionary with the minimum array length for each population.
+    """
+    pop_downsampling_lengths = ht.aggregate(
+        [hl.agg.min(hl.len(ht[f"{prefix}{pop}"])) for pop in pops]
+    )
+
+    pop_lengths = list(zip(pop_downsampling_lengths, pops))
+    logger.info("Found: %s", "".join(map(str, pop_lengths)))
+
+    # check all the arrays of variant counts within a population have the same length
+    assert ht.all(
```

Comment:
I found no circumstance the `skip_assertion` is False, so I delete it. Does the downsampling process have such check as well? If so, we can simplify it.